File: src/main/java/edu/kruchon/natural/lang/parser/codegeneration/KotlinMethod.java
package edu.kruchon.natural.lang.parser.codegeneration;

import edu.kruchon.natural.lang.parser.syntax.Parameter;

public class KotlinMethod {
    final String name;
    final String parameterName;

    public KotlinMethod(String name, Parameter parameter) {
        this.name = name;
        this.parameterName = parameter.getName();
    }

    public String getName() {
        return name;
    }

    public String getParameterClassName() {
        return KotlinGenerationUtils.INSTANCE.firstCharToUpperCase(parameterName);
    }

    public String getParameterName() {
        return parameterName;
    }
}

File: src/main/kotlin/edu/kruchon/natural/lang/parser/codegeneration/KotlinMethod.kt
package edu.kruchon.natural.lang.parser.codegeneration

import edu.kruchon.natural.lang.parser.syntax.Parameter

class KotlinMethod(val name: String, parameter: Parameter) {
    val parameterName: String
    val parameterClassName: String

    init {
        parameterName = parameter.name
        parameterClassName = KotlinGenerationUtils.firstCharToUpperCase(parameterName)
    }
}